tinytodo-go/internal/app/server/entitystore/entityuid.go

```diff
@@ -0,0 +1,114 @@
+package entitystore
+
+import (
+	"code.byted.org/binaryauthorization/tinytodo-go/internal/app/server/entitystore/entitytype"
+	"encoding/json"
+	"fmt"
+	"github.com/cedar-policy/cedar-go"
+	"strings"
+)
+
+// EntityUID is a transparent wrapper around cedar.EntityUID used to represent entities in EntityStore, so that we
+// can define our own UnmarshalJSON and MarshalJSON functions.
```

Comment:
I had to write my own `UnmarshalJSON` method for `EntityUID` (as well as `UserUID`, `ListUID`, `TeamUID` by extension) because I couldn't unmarshal `entities.json`.

Take the user `kesha` as an example:

```json
{
  "users": {
    "User::\"kesha\"": {
      "euid": "User::\"kesha\"",

      // omitted
```

`kesha`'s UID would be `User::\"kesha\"`, and I wanted to unmarshal all users from `entities.json` into a `map[UserUID]*User`. If I did not write my own `UnmarshalJSON` method, I would rely on`cedar.EntityUID`'s `UnmarshalJSON` method which could only unmarshal `{"type":"User","id":"kesha"}`.

In short, `entities.json` does not adhere to [Cedar's Entities and Context syntax](https://docs.cedarpolicy.com/auth/entities-syntax.html), which is why I had to write my own `UnmarshalJSON` method.

I had to write my own `MarshalJSON` method because the Python client (`tinytodo.py`) expects the returned values to be in the format of `User::\"kesha\"` rather than `{"type":"User","id":"kesha"}`.

(I have a feeling that I may have missed an obvious solution or done something that's not idiomatic in Go.)

Reply:
I see. Thank you for clarifying. It does seem like `entities.json` should strive to conform to the published syntax. Maybe that can be addressed in a later PR that merges the two TinyTodo examples.